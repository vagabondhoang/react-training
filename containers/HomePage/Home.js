import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import InputPreview from '../../components/InputPreview';
import setMessage from './actions';
import Button from '../../styles';

type Props = {
  dispatch: (action: Object) => void,
  messageReducer: Object,
};

class Home extends React.Component<Props> {
  static defaultProps = {
    message: 'message',
  };
  handleChange = value => {
    this.props.dispatch(setMessage(value));
  };

  render() {
    const { message } = this.props.messageReducer;

    return (
      <React.Fragment>
        <InputPreview value={message} onChange={this.handleChange} />
        <p>{message}</p>
        <Link to="/about">
          <Button>Go to about page</Button>
        </Link>
      </React.Fragment>
    );
  }
}

export default connect(state => state)(Home);
