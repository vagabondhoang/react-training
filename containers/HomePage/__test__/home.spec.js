import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import Home from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('<Home /> component', () => {
  it('should return correctly', () => {
    const component = shallow(<Home />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
