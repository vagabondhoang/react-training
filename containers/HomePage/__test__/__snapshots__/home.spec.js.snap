// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> component should return correctly 1`] = `
<div>
  Home
  <Link
    replace={false}
    to="/about"
  >
    <styled.button>
      Go about Page
    </styled.button>
  </Link>
  <Link
    replace={false}
    to="/users"
  >
    <styled.button>
      Go Users Page
    </styled.button>
  </Link>
  <h1>
    Reselect Redux
  </h1>
  <Connect(Posts) />
  <Connect(Counter) />
  <h2>
    User 1
  </h2>
  <Connect(PostsByUsers)
    user="user-1"
  />
  <h2>
    User 2
  </h2>
  <Connect(PostsByUsers)
    user="user-2"
  />
</div>
`;
